### PR TITLE
feat: add course reset tab to learner's information

### PR DESCRIPTION
**Pull request by hajorg**
### Summary

https://2u-internal.atlassian.net/browse/AU-1845

Adds Course Reset tab to Learner's Information.
Screenshot 2024-02-23 at 15 45 11

### CODE REVIEW

File: src/users/CourseReset.jsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            Header: intl.formatMessage(messages.recordTableHeaderStatus),
+            accessor: 'status',
+          },
+          {
+            Header: 'Action',
+            accessor: 'action',
+          },
+        ]}
+        data={renderResetData}
+        styleName="custom-table"
```
Comment (leangseu-edx) [inline src/users/CourseReset.jsx:173]: is this needed?

File: src/users/CourseReset.jsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    if (shouldPoll) {
+      intervalId = setInterval(async () => {
+        const data = await getLearnerCourseResetList(username);
+        setCourseResetData(data);
+      }, POLLING_INTERVAL);
+    }
+    return () => clearInterval(intervalId);
+  }, [courseResetData]);
+
+  const handleSubmit = async (courseID) => {
```
Comment (leangseu-edx) [inline src/users/CourseReset.jsx]: write it with `useCallback`.

File: src/users/CourseReset.jsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      });
+      setCourseResetData(updatedCourseResetData);
+    }
+    if (data && data.errors) {
+      setError(data.errors[0].text);
+    }
+    close();
+  };
+
+  useEffect(() => {
```
Comment (leangseu-edx) [inline src/users/CourseReset.jsx]: this `useEffect` should be able to combine.

```js
useEffect(() => {
fetchData();
if (data)
...(above code)
else(error)
}, []);
```

Comment (leangseu-edx) [reply src/users/CourseReset.jsx]: `isMounted` is unnecessary because `[]` already make sure it run only once.

Comment (hajorg) [reply src/users/CourseReset.jsx]: Can you explain what you mean by `this useEffect should be able to combine.`, please?

For the `isMounted` part, I was getting a warning when running test which is why I added it.

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍

File: src/users/CourseReset.jsx (hunk 1 above)
Comment (hajorg) [reply src/users/CourseReset.jsx:173]: Apparently, this is needed for the table style. I'm sorry for the confusion

**Review by Ali-D-Akbar — APPROVED** [review]
Haven't tested this locally but the changes look good to me.

**Pull request merged**